jobserver/views.py

```diff
@@ -26,39 +26,59 @@ class JobDetail(DetailView):
 
 
 class JobList(ListView):
-    model = Job
     ordering = "-pk"
     paginate_by = 25
     template_name = "job_list.html"
 
+    def filter_by_status(self):
+        """
+        Filter JobRequests by possible status
+
+        Status is currently inferred from various Job fields and "bubbled up"
+        to a JobRequest.  This converts the view's QuerySet to a list via a
+        filter which uses properties on JobRequest to cut down the values to
+        those requested.
+
+        TODO: replace with a custom QuerySet once status is driven entirely via
+        the job-runner
+        """
+        status = self.request.GET.get("status")
+
+        if not status:
+            return self.object_list
+
+        status_lut = {
+            "completed": lambda r: r.is_complete,
+            "failed": lambda r: r.is_failed,
```

Comment:
The same point applies here about possibly wanting to distinguish between finished-with-failures and in-progress-with-failures.